secboot/secboot_tpm.go

```diff
@@ -322,15 +323,58 @@ func SealKey(key EncryptionKey, params *SealKeyParams) error {
 		return fmt.Errorf("TPM device is not enabled")
 	}
 
+	pcrProfile, err := buildPCRProtectionProfile(numModels, params.ModelParams)
+	if err != nil {
+		return err
+	}
+
+	// Provision the TPM as late as possible
+	if err := tpmProvision(tpm, params.TPMLockoutAuthFile); err != nil {
+		return err
+	}
+
+	// Seal key to the TPM
+	creationParams := sb.KeyCreationParams{
+		PCRProfile: pcrProfile,
+		PINHandle:  pinHandle,
+	}
+	return sbSealKeyToTPM(tpm, key[:], params.KeyFile, params.TPMPolicyUpdateDataFile, &creationParams)
+}
+
+// ResealKey updates the PCR protection policy for the sealed encryption key according to
+// the specified parameters.
+func ResealKey(params *ResealKeyParams) error {
+	numModels := len(params.ModelParams)
+	if numModels < 1 {
+		return fmt.Errorf("at least one set of model-specific parameters is required")
+	}
+
+	tpm, err := sbConnectToDefaultTPM()
+	if err != nil {
+		return fmt.Errorf("cannot connect to TPM: %v", err)
+	}
+	if !isTPMEnabled(tpm) {
+		return fmt.Errorf("TPM device is not enabled")
+	}
+
+	pcrProfile, err := buildPCRProtectionProfile(numModels, params.ModelParams)
+	if err != nil {
+		return err
+	}
+
+	return sbUpdateKeyPCRProtectionPolicy(tpm, params.KeyFile, params.TPMPolicyUpdateDataFile, pcrProfile)
+}
+
+func buildPCRProtectionProfile(numModels int, modelParams []*SealKeyModelParams) (*sb.PCRProtectionProfile, error) {
```

Comment:
computing length shouldn't be expensive so there's no reason to take numModels, cleaner to just compute len(modelParams) inside again